### ISSUE TITLE
Signature widget in portrait mode

### ISSUE THREAD

**Issue by nap2000** [getodk/collect#794]
#### Software and hardware versions 
Collect v1.4.16

#### Problem description
The current signature widget seems to have gone backwards from its previous capabilities. The phone switches into landscape mode for signature but then only shows a small portrait style window within that to take the signature. End results is that captured signature images are a lot smaller than the used to be and in portrait.

#### Steps to reproduce the problem
Create a form with an image with appearance signature

#### Expected behavior
The screen switches to landscape, the user has nearly the whole length of the phone to do their signature, the captured image is in landscape.

#### Other information 
![device-2017-03-30-105506](https://cloud.githubusercontent.com/assets/1946349/24482472/21d48e40-153d-11e7-9ddf-7d8ec8be2396.jpg)

@max2me,  Maxim may have a view on this issue.

**Comment by rijuldhir** [comment]
I can work on this @lognaturel

**Comment by lognaturel** [comment]
Thanks, @rijuldhir! If you want to just make the usable space fill the screen, go ahead.

Alternately, if you're interested in more fundamentally improving the widget, go ahead and describe what you have in mind here and we can discuss. (It's totally ok to just make the usable space wider and in fact, it might be a good place to start even if you do want to rethink the widget.)

**Comment by nap2000** [comment]
Given this worked a couple of versions ago.  It would be good to get a first fix that undoes the recently introduced problem first.  Any improvements could then be applied as a separate pull request.

**Comment by lognaturel** [comment]
@nap2000 Agreed, that's a good way to proceed. So @rijuldhir, start with getting back to how it worked and then if you're interested, you can file a separate issue to improve it.

**Comment by rijuldhir** [comment]
@lognaturel Will make space fill the screen in this issue first and will be working on improving widget on another PR. Sorry to reply late was just a little busy on GSoC Proposal.

**Comment by lognaturel** [comment]
No problem, @rijuldhir, there's no big rush on this so feel free to do it after your GSoC proposal is submitted!

**Comment by rijuldhir** [comment]
Sure . Thanks @lognaturel

**Comment by lognaturel** [comment]
Well, unless @nap2000 needs a patch for his build now? Otherwise it will be released with the April release.

**Comment by nap2000** [comment]
@ln would you plan to release an update such as this on the 1.4 branch as well as the 1.5 branch?

**Comment by nap2000** [comment]
I should have replied to your question.   No I can wait until the April release.

**Comment by lognaturel** [comment]
We haven't been doing maintenance releases unless there's an ask for specific features/fixes. Do you have devices that were cut out by the API increase? We could certainly do it, especially if you volunteer to prep the release branch! 😬

**Comment by nap2000** [comment]
Hmm.  No won't volunteer to do that :).  Managing the branches in my collect fork is enough.  There are obviously devices that have been cut out by the release although I don't have any specific examples of users.   I know the customer who raised the issue only has recent devices.  I think your approach is fine there should only be a very small number of changes that require an update to 1.4.

**Comment by max2me** [comment]
Ola, checking in a little late, I'm sorry about that. 

Is it happening on every device? If I remember correctly, that activity forces device into landscape orientation and (given current implementation) tries to figure out dimensions of the screen in appropriate orientation to create fitting canvas.

It's possible that on newer devices orientation change happens with different timing and so the calculations will be off with similar symptoms.

**Comment by nap2000** [comment]
The problem occurs on Android 6.0.1

**Comment by lognaturel** [comment]
Thanks for the hint, @max2me. On my Nexus 5, Android 6.0.1, the white area is in fact the width of the phone in portrait rather than the width in landscape. It's just not a big enough difference for me to have noticed something was wrong. I thought the regression might have been introduced by #517 but actually it looks like it was introduced in v1.4.11.

**Comment by lognaturel** [comment]
@rijuldhir This is an important one to get into the April release. Do you think you'll have time to do it by the April 22nd code freeze?

**Comment by rijuldhir** [comment]
@lognaturel Sorry for it I have my exams on and I was working on the long term solution rather than the quick fix. The quick fix is easy just to open widget in Landscape but I have a Pending Pull Request on master which I created unknowingly. So I was waiting for it to get merged.

**Comment by lognaturel** [comment]
That's ok! I just wanted to make sure we have a plan. The quick fix sounds like it will be easy so that's great. You can take your time with the longer-term improvements.

What you can do to work on things in parallel even if you have a PR on your master is to do `git log` to get the commit hash for the latest commit that was in the upstream master (e.g. 8190776384b50a6f3e46311118affd0b5f21bff9) and then create a branch from that with `git branch -b my-new-branch 8190776384b50a6f3e46311118affd0b5f21bff9`.
